sstables/sstables.cc

```diff
@@ -3398,15 +3400,23 @@ future<> sstable::s3_storage::change_state(const sstable& sst, sstring to, gener
 }
 
 future<> sstable::s3_storage::wipe(const sstable& sst) noexcept {
-    auto& sys_ks = sst.manager().system_keyspace();
-
-    co_await sys_ks.sstables_registry_update_entry_status(_location, sst.generation(), status_removing);
-
     co_await coroutine::parallel_for_each(sst._recognized_components, [this, &sst] (auto type) -> future<> {
         co_await _client->delete_object(make_s3_object_name(sst, type));
     });
+}
+
+future<> sstable::s3_storage::delete_with_system_keyspace(std::vector<shared_sstable> ssts) {
+    co_await coroutine::parallel_for_each(ssts, [] (shared_sstable sst) -> future<> {
+        const s3_storage* storage = dynamic_cast<const s3_storage*>(&sst->get_storage());
+        if (!storage) {
+            on_fatal_internal_error(sstlog, "Atomically deleted sstables must be of same storage type");
+        }
 
-    co_await sys_ks.sstables_registry_delete_entry(_location, sst.generation());
+        auto& sys_ks = sst->manager().system_keyspace();
+        co_await sys_ks.sstables_registry_update_entry_status(storage->_location, sst->generation(), status_removing);
+        co_await sst->unlink();
```

Comment:
how do we handle entries with removing status on boot?

note it still suffers with lack of atomic deletion, so worth adding a FIXME.

example:
let's say sstable A and B are compacted together.
A has a tombstone for key K
B has a data shadowed by tombstone in A.
we have to make sure A and B are deleted atomically. 
otherwise, deletion of A but B will result in data in B being resurrected.

for implementing actual atomic deleiton, we need to batch the status update for all sstables being deleted together., and only then proceed to wipe the entries.

so it cannot happen that A is updated as removing, but B isn't, if there's a crash midway.

I think we can start by splitting status removing update and wipe, into prepare and execution phases, respectively.

Reply:
> how do we handle entries with removing status on boot?

Currently we just skip them, #13024 should fix that